executor/src/witgen/generator.rs

```diff
@@ -44,11 +44,11 @@ impl<'a, T: FieldElement> CompletableIdentities<'a, T> {
     }
 }
 
-pub struct Generator<'a, T: FieldElement, QueryCallback: Send + Sync> {
+pub struct Generator<'a, 'b, T: FieldElement, QueryCallback: Send + Sync> {
```

Comment:
Oh, yeah, I think `IdentityProcessor` (which we'd have once per machine) could just own it then. Actually, that's the reason why `BlockMachine` doesn't own it's identity processor (like`Generator`) but instantiates it every time it is called (at which time it will have a mutable reference to `FixedLookup`).